S25 test environment setup and documentation for writing test (#1221)
gordon-cs/gordon-360-api: Gordon360.Tests/ReadMe.md

4. [Authentication Setup for Test Purposes](#4-authentication-setup-for-test-purposes-)
5. [Actual Test Demo](#5-actual-test-demo-)
6. [ChatGPT Prompt Tips for Writing Tests](#6-chatgpt-prompt-tips-for-writing-tests-)
7. [Test Dubugging Tips](#7-test-dubugging-tips-)
8. [Automating Tests with GitHub Actions and Git Hooks](#8-automating-tests-with-github-actions-and-git-hooks-)

---

## 1. Setting Up Test Environment 🛠️  

To set up the test project in **Gordon360** from scratch:
Remeber that Step1 to Step3 are only needed if you haven't set up the test project yet. If you already have a test project, skip to Step 4 to start.

### Step 1: Create the Test Project

If not already created, create a new test project:

```bash
dotnet new xunit -n Gordon360.Tests
```

### Step 2: Reference Main Project

Ensure the test project references the main Gordon360 project:

```bash
cd Gordon360.Tests
dotnet add reference ../Gordon360/Gordon360.csproj
```

### Step 3: Install Required NuGet Packages

Install the following packages:

```bash
dotnet add package Moq

dotnet add package Microsoft.AspNetCore.Mvc

dotnet add package Microsoft.AspNetCore.Http
```

### Step 4: Folder Structure

> 

Reviewer: ```suggestion
Below we show how to create a test for the Profiles Controller.  The files associated with the tests need to be organized in a folder hierarchy under `Gordon360.Tests`.  In the case of Profiles Controller tests the hierarchy should look like
```

> Make sure your test file is in the appropriate folder:

Reviewer: ```suggestion
```

```
Gordon360.Tests/

> ├── Controllers_Test/ create this folder only if not exists

Reviewer: ```suggestion
├── Controllers_Test/   # create this folder if it does not exist
```

│   └── ProfilesControllerTests.cs # or any other controller you are testing
```

### Step 5: Import Namespaces for Test Files

Ensure your the namespace needed in your test file are included in "TestUsings.cs":

> // ✅ Global using directives make these namespaces available across all test files

Reviewer: Should these lines be in the code block below?

Author: ok comment character is removed.
